src/language-css/printer-postcss.js

```diff
@@ -461,166 +474,142 @@ function genericPrint(path, options, print) {
-          isControlDirective && isIfElseKeywordNode(iNextNode);
-        const isEachKeyword = isControlDirective && isEachKeywordNode(iNode);
-        const isNextEachKeyword =
-          isControlDirective && isEachKeywordNode(iNextNode);
-        const isForKeyword =
+        // Ignore inline comment, they already contain newline at end (i.e. `// Comment`)
+        // Add `hardline` after inline comment (i.e. `// comment\n foo: bar;`)
+        const isInlineComment = isInlineValueCommentNode(iNode);
+
+        if (
+          (iPrevNode && isInlineValueCommentNode(iPrevNode)) ||
+          isInlineComment ||
+          isInlineValueCommentNode(iNextNode)
+        ) {
+          if (isInlineComment) {
+            parts.push(hardline);
+          }
+
+          continue;
+        }
+
+        // Handle keywords in SCSS control directive
+        if (
+          isControlDirective &&
+          (isEqualityOperatorNode(iNextNode) ||
+            isRelationalOperatorNode(iNextNode) ||
+            isIfElseKeywordNode(iNextNode) ||
+            isEachKeywordNode(iNode) ||
+            isForKeywordNode(iNode))
+        ) {
+          parts.push(" ");
+
+          continue;
+        }
+
+        // At-rule `namespace` should be in one line
+        if (
           atRuleAncestorNode &&
-          atRuleAncestorNode.name === "for" &&
-          isForKeywordNode(iNode);
-        const isNextForKeyword =
-          isControlDirective && isForKeywordNode(iNextNode);
-        const IsNextColon = iNextNode.value === ":";
+          atRuleAncestorNode.name.toLowerCase() === "namespace"
+        ) {
+          parts.push(" ");
 
+          continue;
+        }
+
+        // Formatting `grid` property
         if (isGridValue) {
           if (iNode.source.start.line !== iNextNode.source.start.line) {
             parts.push(hardline);
+
             didBreak = true;
           } else {
             parts.push(" ");
           }
-        } else if (iNode.type === "value-comment" && iNode.inline) {
-          parts.push(hardline);
-        } else if (
-          isNextMathOperator ||
-          isNextEqualityOperator ||
-          isNextRelationalOperator ||
-          isNextIfElseKeyword ||
-          isForKeyword ||
-          isEachKeyword ||
-          (atRuleAncestorNode &&
-            atRuleAncestorNode.name.toLowerCase() === "namespace")
-        ) {
+
+          continue;
+        }
+
+        // Add `space` before next math operation
+        // Note: `grip` property have `/` delimiter and it is not math operation, so
+        // `grid` property handles above
+        if (isNextMathOperator) {
           parts.push(" ");
-        } else if (
-          !IsNextColon ||
-          isEqualityOperator ||
-          isRelationalOperator ||
-          isNextForKeyword ||
-          isNextEachKeyword
-        ) {
-          parts.push(line);
+
+          continue;
         }
+
+        // Be default all values go through `line`
+        parts.push(line);
```

Comment:
Just add own logic before this for formatting and it is all, easy read = easy maintain